File: Array/problems/easy/find_missing.java
public class find_missing {
  public static int FindMissing(int arr[]) {
    int sum = 0;
    int largest = arr[0];
    for (int i = 0; i < arr.length; i++){
      if (largest < arr[i]){
        largest = arr[i];
      }
      sum += arr[i];
    }

    return (largest * (largest+1)/2) - sum;
  }
  public static void main(String[] args) {
    System.out.println("[Find Missing Element]\nTime Comp: O(N)");
    int arr[] = { 1, 3, 7, 5, 6, 2 };
    int missing = FindMissing(arr);
    System.out.println("The missing element is " + missing);
  }
}

File: Array/problems/easy/find_missing.rs
fn main() {
    println!("[Find Missing Element]\nTime Comp: O(N)");
    let arr = [1, 3, 7, 5, 6, 2];
    let missing = find_missing(&arr);
    println!("The missing number is {}", missing);
}

fn find_missing(arr: &[i32]) -> i32 {
    let mut sum = 0;
    let mut largest = arr[0];
    for (_, &val) in arr.iter().enumerate() {
        if largest < val {
            largest = val;
        }
        sum += val;
    }
    (largest * (largest + 1) / 2) - sum
}
